librz/bin/format/luac/luac_common.h

```diff
@@ -0,0 +1,251 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// SPDX-FileCopyrightText: 2021 Heersin <teablearcher@gmail.com>
+
+// put common definition of luac
+
+#ifndef BUILD_LUAC_COMMON_H
+#define BUILD_LUAC_COMMON_H
+
+#include <rz_bin.h>
+#include <rz_lib.h>
+#include <rz_list.h>
+
+typedef ut32 LUA_INSTRUCTION;
+
+/* Macros About Luac Format */
+#define LUAC_MAGIC_OFFSET   0x00
+#define LUAC_MAGIC_SIZE     4
+#define LUAC_VERSION_OFFSET 0x04
+#define LUAC_VERSION_SIZE   1
+
+#define LUAC_MAGIC "\x1b\x4c\x75\x61"
+
+/* Lua Constant Tag */
+#define makevariant(t, v) ((t) | ((v) << 4))
+
+#define LUA_TNIL     0
+#define LUA_TBOOLEAN 1
+#define LUA_TNUMBER  3
+#define LUA_TSTRING  4
+
+#define LUA_VNIL    makevariant(LUA_TNIL, 0)
+#define LUA_VFALSE  makevariant(LUA_TBOOLEAN, 0)
+#define LUA_VTRUE   makevariant(LUA_TBOOLEAN, 1)
+#define LUA_VNUMINT makevariant(LUA_TNUMBER, 0) /* integer numbers */
+#define LUA_VNUMFLT makevariant(LUA_TNUMBER, 1) /* float numbers */
+#define LUA_VSHRSTR makevariant(LUA_TSTRING, 0) /* short strings */
+#define LUA_VLNGSTR makevariant(LUA_TSTRING, 1) /* long strings */
+
+/* Solution A: only store offset info about different parts
```

Comment:
Please use doxygen comments to comment data structures and API functions. See rz_cmd.h and cmd_api.c for the style we usually use.